src/baseline.rs

```diff
@@ -7,22 +7,210 @@ use crate::dump::RustDocCommand;
+    ManifestPath { path: &'a Path, name: String },
+}
+
+/// To get the rustdoc of the baseline, we first create a placeholder project somewhere
+/// with the baseline as a dependency, and run `cargo rustdoc` on it.
+fn create_placeholder_rustdoc_manifest(
+    crate_source: &CrateSource,
+) -> anyhow::Result<cargo_toml::Manifest<()>> {
+    use cargo_toml::*;
+
+    Ok(Manifest::<()> {
+        package: {
+            let mut package = Package::new("rustdoc", "0.0.0");
+            package.publish = Inheritable::Set(Publish::Flag(false));
+            Some(package)
+        },
+        workspace: Some(Workspace::<()>::default()),
+        lib: {
+            let product = Product {
+                path: Some("lib.rs".to_string()),
+                ..Product::default()
+            };
+            Some(product)
+        },
+        dependencies: {
+            let project_with_features: DependencyDetail = match crate_source {
+                CrateSource::Registry { crate_ } => DependencyDetail {
+                    // We need the *exact* version as a dependency, or else cargo will
+                    // give us the latest semver-compatible version which is not we want.
+                    // Fixes: https://github.com/obi1kenobi/cargo-semver-checks/issues/261
+                    version: Some(format!("={}", crate_.version())),
+                    features: crate_features::get_all_crate_features_from_registry(crate_),
+                    ..DependencyDetail::default()
+                },
+                CrateSource::ManifestPath { path, .. } => DependencyDetail {
+                    path: Some(
+                        // The manifest will be saved in some other directory,
+                        // so for convenience, we're using absolute paths.
+                        path.canonicalize()
+                            .with_context(|| "failed to canonicalize manifest path")?
+                            .to_str()
+                            .with_context(|| "manifest path is not valid UTF-8")?
+                            .to_string(),
+                    ),
+                    features: crate_features::get_all_crate_features_from_manifest(path),
+                    ..DependencyDetail::default()
+                },
+            };
+            let mut deps = DepsSet::new();
+            deps.insert(
+                match crate_source {
+                    CrateSource::Registry { crate_ } => crate_.name().to_string(),
+                    CrateSource::ManifestPath { name, path: _ } => name.to_string(),
+                },
+                Dependency::Detailed(project_with_features),
+            );
+            deps
+        },
+        ..Default::default()
+    })
+}
+
+fn save_placeholder_rustdoc_manifest(
+    placeholder_build_dir: &Path,
+    placeholder_manifest: cargo_toml::Manifest<()>,
+) -> anyhow::Result<PathBuf> {
+    std::fs::create_dir_all(placeholder_build_dir).with_context(|| "failed to create build dir")?;
```

Comment:
If you don't allocate, you can just use `.context`. The `with` variant is for when you allocate to only do it in case of an error.

Reply:
Done in ff7c57118d70ef99a155b120885e933d0299be2d. Do I understand correctly that e.g. `.with_context(|| anyhow::format_err!("{} not found in registry", name))` allocates, so it should be in `with_context`?

Reply:
Yes!